tools/export_all_kernel_insight.py

```
@@ -0,0 +1,665 @@
+    if failed:
+        log("FAILED_FUNCS " + ",".join(f"{r.get('func')}:{r.get('status')}" for r in failed))
+        return 2
+    return 0
+
+
+if __name__ == "__main__":
+    try:
+        raise SystemExit(main())
+    except StepError as exc:
+        print(f"error: {exc}", file=sys.stderr)
+        raise SystemExit(1)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)